### PR TITLE
replace AkkaSpec with StreamSpec for stream tests

### PR DESCRIPTION
**Pull request by Hawstein**
Ref #21011

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST, not sure if it makes sense everywhere. Let's see

**Comment by ktoso** [comment]
OK TO TEST

**Comment by Hawstein** [comment]
@ktoso I just replace `AkkaSpec` with `StreamSpec` in `akka-stream-tests` package, not everywhere. IMO, `StreamSpec` is just an enhancement of `AkkaSpec`, so it's OK to use it in `akka-stream-tests`.

**Comment by ktoso** [comment]
I meant "everywhere" as "everywhere in akka streams" :)
Validator is happy so I think we're good here, thanks!

LGTM

File: akka-testkit/src/test/scala/akka/testkit/AkkaSpec.scala [hunk 1]
```diff
@@ -44,7 +44,7 @@ object AkkaSpec {
 
   def getCallerName(clazz: Class[_]): String = {
     val s = (Thread.currentThread.getStackTrace map (_.getClassName) drop 1)
-      .dropWhile(_ matches "(java.lang.Thread|.*AkkaSpec.?$)")
+      .dropWhile(_ matches "(java.lang.Thread|.*AkkaSpec.?$|.*StreamSpec.?$)")
```
Comment (ktoso) [inline akka-testkit/src/test/scala/akka/testkit/AkkaSpec.scala]: good, thanks

**Comment by ktoso** [comment]
LGTM

**Comment by ktoso** [comment]
PLS BUILD

**Comment by johanandren** [comment]
LGTM, thanks!

**Pull request merged**